src/world/common/UnkFunc29.inc.c

```diff
@@ -0,0 +1,45 @@
+#include "common.h"
+#include "map.h"
+
+ApiStatus N(UnkFunc29)(ScriptInstance *script, s32 isInitialCall) {
+    s32 i;
+
+    if (isInitialCall) {
+        set_background_color_blend(0, 0, 0, 0xFF);
+        script->functionTemp[0].s = 0xFF;
+        script->functionTemp[1].s = 0;
+    }
+
+    script->functionTemp[0].s -= 8;
+    if (script->functionTemp[0].s < 0) {
+        script->functionTemp[0].s = 0;
+    }
+
+    set_background_color_blend(0, 0, 0, script->functionTemp[0].s);
+    if ((script->functionTemp[0].s == 0) && (script->functionTemp[1].s == 0)) {
+        script->functionTemp[1].s = 1;
+    } else if (script->functionTemp[1].s == 1) {
+        func_8011D82C(0);
+        *D_801512F0 = 0;
+
+        for (i = 0; i < 64; i++) {
+            Npc* npc = get_npc_by_index(i);
+            if (npc != NULL && npc->flags != 0) {
+                if (npc->npcID != -4 && npc->npcID != UNK_ALPHA_FUNC_NPC) {
+                    npc->flags &= 0xFF7FFFFF;
+                }
+            }
+        }
+
+        for (i = 0; i < 256; i++) {
```

Comment:
Possibly 256 should be `ARRAY_COUNT(itemEntities)`, wherever that is?

Reply:
I'm not sure what that is. There certianly aren't 256 item entities if you mean as defined by the script. It seems more like a constant maximum possible item entities?

Reply:
Yeah, perhaps worth adding it to macros.h